### PR TITLE
Use 'unsigned short' instead of 'unsigned'.

### PR DESCRIPTION
**Pull request by TIHan**
Should resolve work item 1507451.

### CODE REVIEW

File: src/coreclr/ildasm/dres.cpp [hunk 1]
```diff
@@ -155,7 +155,7 @@ DWORD   DumpResourceToFile(_In_ __nullterminated WCHAR*   wzFileName)
     DWORD ret = 0;
     DWORD   dwResDirRVA;
     DWORD   dwResDirSize;
-    unsigned ulNumResNodes=0;
+    unsigned short ulNumResNodes=0;
```
Comment (tannergooding) [inline src/coreclr/ildasm/dres.cpp:158]: You might also be able to use `uint16_t` which is a bit more explicit

**Review by jakobbotsch — COMMENTED** [review]
Is there a 2^16 limit on resources? Would it be better to change the induction variable below to be `unsigned` instead?

**Comment by TIHan** [comment]
> Is there a 2^16 limit on resources? Would it be better to change the induction variable below to be unsigned instead?

I thought of that as well, it's unclear since 'unsigned short' is used multiple times in the body.

File: src/coreclr/ildasm/dres.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/ildasm/dres.cpp:158]: Do we use 'uint16_t' commonly? 'unsigned short' is used multiple times in the body of this function.

**Comment by TIHan** [comment]
I think to be safe I will change it to use unsigned for the whole thing if it's possible.

File: src/coreclr/ildasm/dres.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/ildasm/dres.cpp:158]: It depends on the code you're looking at and how new it is.

`unsigned short` doesn't have a "well-defined" size in C and just happens to currently be the same as `uint16_t` on all platforms we currently target.

`uint16_t` is always going to be 16-bits on all platforms and so its often better to be explicit where feasible (remove ambiguity, reduce potential bugs, etc).

**Comment by TIHan** [comment]
@jakobbotsch I changed the 'unsigned short' to 'unsigned' - this should be safe to do.

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**